renderer/src/main.tsx

```diff
@@ -15,8 +16,19 @@ ReactDOM.createRoot(
   </React.StrictMode>
 )
 
-window.electron.onActivityLogged(entry => {
-  console.log('[ACTIVITY] %j', entry)
+const activities: Activity[] = []
+
+window.electron.onActivityLogged(activity => {
+  activities.push(activity)
+  activities.sort((a, b) => {
+    return a.timestamp !== b.timestamp
+      ? b.timestamp - a.timestamp
+      : a.id.localeCompare(b.id)
```

Comment:
Are you sure ids created by `crypto.randomUUID()` monotonically increasing over time?

Maybe the order of events reported in the same millisecond does not matter and that's why we can fallback our sorting comparisons to comparing ids?

I think it would be great to add a code comment to clarify this.

Reply:
They don't increase, and that's not an issue in my mind. As long as we sort deterministically, no human will have noticed that the events from the same millisecond have been mixed up. And we sort deterministically by giving events IDs when storing them. I'll add a comment 👍

Reply:
alternative idea: Is there a timing api with such precision that it's guaranteed that two successive calls won't get the same values?

Reply:
`process.hrtime()` _seems_ to work:

```
> [process.hrtime(), process.hrtime()]
[ [ 845635, 876114697 ], [ 845635, 876117208 ] ]
```

Reply:
> As long as we sort deterministically, no human will have noticed that the events from the same millisecond have been mixed up. 

Unless these two events happen to be something like "Starting the station" and "Starting Saturn module". It would look weird to have them in the reverse direction.

Anyhow, I think this does not really matter, especially not now.